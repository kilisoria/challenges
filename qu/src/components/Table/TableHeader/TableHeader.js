import React, { memo } from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';

const TableHeader = ({ classes, order, orderBy, numSelected, rowCount, onRequestSort }) => {
  const createSortHandler = (property) => (event) => {
    onRequestSort(event, property);
  };
    
const HEAD_CELLS = [
  { id: 'name', numeric: false, disablePadding: true, label: 'Name' },
  { id: 'gravity', numeric: false, disablePadding: false, label: 'Gravity' },
  { id: 'terrain', numeric: false, disablePadding: false, label: 'Terrain' },
  { id: 'climate', numeric: false, disablePadding: false, label: 'Climate' },
  { id: 'population', numeric: true, disablePadding: false, label: 'Population' },
];

const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: '#75CFDB',
    color: theme.palette.common.black,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);
    
  return (
    <TableHead>
      <TableRow>
        <StyledTableCell padding="checkbox" />
        {HEAD_CELLS.map((headCell) => (
          <StyledTableCell
            key={headCell.id}
            align='right'
            padding={headCell.disablePadding ? 'none' : 'default'}
            sortDirection={orderBy === headCell.id ? order : false}
          >
            <TableSortLabel
              active={orderBy === headCell.id}
              direction={orderBy === headCell.id ? order : 'asc'}
              onClick={createSortHandler(headCell.id)}
            >
              {headCell.label}
              {orderBy === headCell.id ? (
                <span className={classes.visuallyHidden}>
                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
                </span>
              ) : null}
            </TableSortLabel>
          </StyledTableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}

TableHeader.propTypes = {
  classes: PropTypes.object.isRequired,
  numSelected: PropTypes.number.isRequired,
  onRequestSort: PropTypes.func.isRequired,
  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
  orderBy: PropTypes.string.isRequired,
  rowCount: PropTypes.number.isRequired,
};

export default memo(TableHeader);