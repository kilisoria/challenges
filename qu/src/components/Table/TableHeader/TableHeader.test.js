import React from 'react';
import renderer from 'react-test-renderer';

import TableHeader from './TableHeader';

describe('GIVEN TableHeader component', () => {
  describe('WHEN renders by default', () => {
    it('THEN should match the snapshot', () => {
      const tree = renderer.create(<TableHeader
        order="asc"
        orderBy="name"
        rowCount={10}
        classes={{}}
        numSelected={1}
        onRequestSort={() => { }} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});