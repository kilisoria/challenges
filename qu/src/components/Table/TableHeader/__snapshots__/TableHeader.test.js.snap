// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN TableHeader component WHEN renders by default THEN should match the snapshot 1`] = `
<thead
  className="MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-paddingCheckbox"
      scope="col"
    />
    <th
      aria-sort="ascending"
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-alignRight MuiTableCell-paddingNone"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Name
        <span>
          sorted ascending
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Gravity
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Terrain
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Climate
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 MuiTableCell-alignRight"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Population
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
  </tr>
</thead>
`;
